node/src/components/consensus/highway_core/highway_testing.rs

```diff
@@ -215,20 +215,16 @@ impl HighwayValidator {
         Ok(self.finality_detector.run(&self.highway)?.collect())
     }
 
-    fn post_hook(&mut self, rng: &mut NodeRng, msg: HighwayMessage) -> Vec<HighwayMessage> {
+    fn post_hook(
+        &mut self,
+        rng: &mut NodeRng,
+        delivery_time: Timestamp,
+        msg: HighwayMessage,
+    ) -> Vec<HighwayMessage> {
         match self.fault.as_ref() {
-            None => {
-                // Honest validator.
-                match &msg {
-                    HighwayMessage::NewVertex(_)
-                    | HighwayMessage::Timer(_)
-                    | HighwayMessage::RequestBlock(_) => vec![msg],
-                    HighwayMessage::WeAreFaulty(ev) => {
-                        panic!("validator equivocated unexpectedly: {:?}", ev);
-                    }
-                }
-            }
-            Some(DesFault::Mute) => {
+            Some(DesFault::Mute(from, till))
+                if *from <= delivery_time && delivery_time <= *till =>
```

Comment:
Please add a comment explaining what this condition means, and why it has to have a special handling.